mmtk/src/julia_scanning.rs

```diff
@@ -422,15 +490,24 @@ pub fn process_slot<EV: SlotVisitor<JuliaVMSlot>>(closure: &mut EV, slot: Addres
     #[cfg(debug_assertions)]
     {
         use crate::JuliaVM;
+        use log::warn;
         use mmtk::vm::slot::Slot;
 
-        if let Some(objref) = simple_slot.load() {
-            debug_assert!(
-                mmtk::memory_manager::is_in_mmtk_spaces::<JuliaVM>(objref),
-                "Object {:?} in slot {:?} is not mapped address",
-                objref,
-                simple_slot
+        if PRINT_OBJ_TYPE {
+            println!(
+                "\tprocess slot = {:?} - {:?}\n",
+                simple_slot,
+                simple_slot.load()
             );
+        }
+
+        if let Some(objref) = simple_slot.load() {
+            if !mmtk::memory_manager::is_in_mmtk_spaces::<JuliaVM>(objref) {
```

Comment:
This is now always executed. It was a debug assertion.

Reply:
Rolled back that change. It was just for debugging purposes.